hexrd/ui/image_canvas.py

```diff
@@ -233,6 +248,28 @@ def draw_detector_borders(self):
 
         self.draw()
 
+    def reformat_overlays(self, data):
+        x, y = data[:, 1], data[:, 0]
+        if self.mode == 'cartesian':
+            # It's centered on (0, 0). Recenter it in place.
+            self.recenter_points((0, 0), x, y)
+        elif self.mode == 'polar':
+            # Swap axes and convert to degrees.
+            x, y = np.degrees(y), np.degrees(x)
+            # It's centered on 180. Recenter on 0.
+            y -= 180.0
+
+        return x, y
+
+    def recenter_points(self, old_center, x, y):
+        # Recenters the points in place
+        new_extent = self.axes_images[0].get_extent()
+        new_center = ((new_extent[1] + new_extent[0]) / 2.0,
+                      (new_extent[3] + new_extent[2]) / 2.0)
+
+        x += new_center[0] - old_center[0]
+        y += new_center[1] - old_center[1]
```

Comment:
Same here; the coordinate system doesn't change.  Not sure why we need to recenter?  Or is this on mode change?  I noticed some weird behavior with the overlays in the current master with them moving around.  What precipitated the need to "recenter"?

Reply:
Yeah, this is what I get without the re-centering:

![cartesian](https://user-images.githubusercontent.com/9558430/84192099-818bd600-aa67-11ea-926b-e6dffff7f97e.png)

See that the rings are centered on (0, 0), rather than in the center of the image.

If our Cartesian viewer should be centered on (0, 0) as well, I can look into why it isn't doing that. Otherwise, I might be able to find a better fix for moving the overlays. I agree that this might cause some strange issues where the overlays move when we change parameters.

Reply:
Okay, I have some more info and I think I figured out a better way to do it.

Looks like it might be more ideal for me to do it this way:

1. Always convert Cartesian overlays to pixel coordinates in the overlay class (only raw overlays were being converted to pixel coordinates before).
2. Rescale the Cartesian overlays to match the rescaling of the Cartesian image.

The reason 2 is necessary, is because we are rescaling the extents of the Cartesian image to correct physical units in millimeters (#186). Thus, we need to rescale the Cartesian overlays as well.

Reply:
This should be fixed in 0c5caae1567f377449c0c4d7ea407b567e49bed0. Let me know if there are any issues with it.